Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -13139,10 +13139,14 @@ void WebPageProxy::substitutionsPanelIsShowing(CompletionHandler<void(bool)>&& c
     completionHandler(TextChecker::substitutionsPanelIsShowing());
 }
 
-void WebPageProxy::showCorrectionPanel(AlternativeTextType panelType, const FloatRect& boundingBoxOfReplacedString, const String& replacedString, const String& replacementString, const Vector<String>& alternativeReplacementStrings)
+void WebPageProxy::showCorrectionPanel(AlternativeTextType panelType, const FloatRect& boundingBoxOfReplacedString, const String& replacedString, const String& replacementString, const Vector<String>& alternativeReplacementStrings, std::optional<FrameIdentifier> rootFrameID)
 {
-    if (RefPtr pageClient = this->pageClient())
-        pageClient->showCorrectionPanel(panelType, boundingBoxOfReplacedString, replacedString, replacementString, alternativeReplacementStrings);
+    if (RefPtr pageClient = this->pageClient()) {
+        convertRectToMainFrameCoordinates(boundingBoxOfReplacedString, rootFrameID, [pageClient = WTFMove(pageClient), panelType = WTFMove(panelType), boundingBoxOfReplacedString = WTFMove(boundingBoxOfReplacedString), replacedString = WTFMove(replacedString),
+            replacementString = WTFMove(replacementString), alternativeReplacementStrings = WTFMove(alternativeReplacementStrings)](std::optional<WebCore::FloatRect> correctedBoundingBox) {
+            pageClient->showCorrectionPanel(panelType, correctedBoundingBox ? correctedBoundingBox.value() : boundingBoxOfReplacedString, replacedString, replacementString, alternativeReplacementStrings);
```

Comment:
If correctedBoundingBox is nullopt then we should early return.  That means the frame is gone or the process has crashed, and either way we don't need to show anything.
I'd also rename correctedBoundingBox to convertedBoundingBox